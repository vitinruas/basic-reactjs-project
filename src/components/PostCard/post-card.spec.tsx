import { render, screen } from '@testing-library/react'
import { PostCard } from '.'
import IPost from '../../interfaces/post-interface'

const fakePost: IPost = {
  id: 1,
  title: 'any_title',
  body: 'any_body',
  photo: 'any_photo',
  userId: 2,
}

describe('<PostCard />', () => {
  test('Must render PostCard correctly', () => {
    render(<PostCard post={fakePost} />)

    const img = screen.getByRole('img', { name: fakePost.title })
    const title = screen.getByRole('heading', { name: fakePost.title })
    const paragraph = screen.getByText(/any_body/)

    expect(img).toHaveAttribute('src', fakePost.photo)
    expect(title).toBeInTheDocument()
    expect(paragraph).toBeInTheDocument()
  })

  test('Must match snapshot', () => {
    const { container } = render(<PostCard post={fakePost} />)
    expect(container['firstChild']).toMatchSnapshot()
  })
})
