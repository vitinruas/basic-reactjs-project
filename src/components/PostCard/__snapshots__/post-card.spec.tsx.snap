// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard /> Must match snapshot 1`] = `
<div
  class="post"
>
  1
  <picture
    class="post-image"
  >
    <img
      alt="any_title"
      src="any_photo"
    />
  </picture>
  <div
    class="post-content"
  >
    <h1>
      any_title
    </h1>
    <p>
      Description:
      any_body
    </p>
  </div>
</div>
`;
